ui/page/history/index.js

```diff
@@ -0,0 +1,59 @@
+import { connect } from 'react-redux';
+
+import { withRouter } from 'react-router-dom';
+import HistoryPage from './view';
+import {
+  selectTitleForUri,
+  selectClaimIsMine,
+  makeSelectClaimIsPending,
+  makeSelectClaimForClaimId,
+  makeSelectChannelForClaimUri,
+} from 'redux/selectors/claims';
+
+import {
+  makeSelectCollectionForId,
+  makeSelectUrlsForCollectionId,
+  makeSelectIsResolvingCollectionForId,
+  makeSelectCollectionIsMine,
+  makeSelectCountForCollectionId,
+  makeSelectEditedCollectionForId,
+} from 'redux/selectors/collections';
+
+import { getThumbnailFromClaim } from 'util/claim';
+import { doFetchItemsInCollection, doCollectionDelete, doCollectionEdit } from 'redux/actions/collections';
+import { selectUser } from 'redux/selectors/user';
+
+const select = (state, props) => {
+  const { match } = props;
+  const { params } = match;
+  const { collectionId } = params;
+
+  const claim = collectionId && makeSelectClaimForClaimId(collectionId)(state);
+  const uri = (claim && (claim.canonical_url || claim.permanent_url)) || null;
+
+  return {
+    collectionId,
+    claim,
+    collection: makeSelectCollectionForId(collectionId)(state),
+    collectionUrls: makeSelectUrlsForCollectionId(collectionId)(state),
+    collectionCount: makeSelectCountForCollectionId(collectionId)(state),
+    isResolvingCollection: makeSelectIsResolvingCollectionForId(collectionId)(state),
+    title: selectTitleForUri(state, uri),
+    thumbnail: getThumbnailFromClaim(claim),
+    isMyClaim: selectClaimIsMine(state, claim), // or collection is mine?
+    isMyCollection: makeSelectCollectionIsMine(collectionId)(state),
+    claimIsPending: makeSelectClaimIsPending(uri)(state),
```

Comment:
Most of the "collection" stuff is not necessary or not used, including the "perform" stuff below.
I think it will be troublesome trying to sync up with an actual collection/playlist.
`uri=history` is all that we need here.